src/main/java/com/example/nasastealphoto/service/NasaService.java

```diff
@@ -0,0 +1,75 @@
+package com.example.nasastealphoto.service;
+
+import com.example.nasastealphoto.exception.NasaUrlProcessingException;
+import com.example.nasastealphoto.model.Camera;
+import com.example.nasastealphoto.model.Picture;
+import com.example.nasastealphoto.util.UrlCreator;
+import com.fasterxml.jackson.databind.JsonNode;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.StreamSupport;
+
+@Service
+@RequiredArgsConstructor
+public class NasaService {
+    public static final String PHOTOS = "photos";
+    public static final String CAMERA = "camera";
+    public static final String ID = "id";
+    public static final String IMG_SRC = "img_src";
+    public static final String NAME = "name";
+    private final RestTemplate restTemplate;
+    private final CameraService cameraService;
+    private final UrlCreator urlCreator;
+
+    public void saveData(String sol) {
```

Comment:
@romanovosad87 it seems like there are no checks for the existing data. In other words, if I run the same request twice, it will call NASA and will try to store the same data until it fails. As NASA stored pictures by sol, it makes sense to follow their approach and **check by which `sol` pictured are already stolen**.